aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -2874,8 +2874,10 @@ static void lu_solve_looped_magma(const Tensor& b, const Tensor& lu, const Tenso
 static void lu_solve_dispatch(const Tensor& b, const Tensor& lu, const Tensor& pivots) {
   auto batch_size = batchCount(lu);
   auto m = lu.size(-2);
+  auto b1 = b.size(-2);
+  auto b2 = b.size(-1);
 #ifdef USE_CUSOLVER
-  if (batch_size == 1 && m > 512) {
+  if ((batch_size == 1 && m > 512) || (batch_size <= 8 && (m > 1024 || b1 > 1024 || b2 > 1024))) {
```

Comment:
I think this PR overall looks good. Since this boolean `(m > 1024 || b1 > 1024 || b2 > 1024)` is reused below, we can assign it to a variable just below the `auto b2 = ...` line.

Reply:
Thanks, updated.